mmv1/templates/terraform/custom_expand/secret_version_access.go.tmpl

```diff
@@ -0,0 +1,55 @@
+{{/*
+	The license inside this block applies to this file
+	Copyright 2025 Google Inc.
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+*/ -}}
+func expand{{$.GetPrefix}}{{$.TitlelizeProperty}}(v interface{}, d tpgresource.TerraformResourceData, config *transport_tpg.Config) (interface{}, error) {
+	transformed := make(map[string]interface{})
+	transformedSecretData, err := expandSecretManagerSecretVersionPayloadSecretData(d.Get("secret_data"), d, config)
+	if err != nil {
+		return nil, err
+	} else if val := reflect.ValueOf(transformedSecretData); val.IsValid() && !tpgresource.IsEmptyValue(val) {
+		transformed["data"] = transformedSecretData
+	}
+
+{{- if ne $.ResourceMetadata.Compiler "terraformgoogleconversion-codegen" }}
+	transformedSecretDataWo, err := expandSecretManagerSecretVersionPayloadSecretDataWo(d.Get("secret_data_wo"), d.(*schema.ResourceData), config)
+	if err != nil {
+		return nil, err
+	} else if val := reflect.ValueOf(transformedSecretDataWo); val.IsValid() && !tpgresource.IsEmptyValue(val) {
+		transformed["data"] = transformedSecretDataWo
+	}
+{{- end }}
+	return transformed, nil
+}
+
+func expandSecretManagerSecretVersionPayloadSecretData(v interface{}, d tpgresource.TerraformResourceData, config *transport_tpg.Config) (interface{}, error) {
+	if v == nil {
+		return nil, nil
+	}
+
+	if d.Get("is_secret_data_base64").(bool) {
+		return v, nil
+	}
+	return base64.StdEncoding.EncodeToString([]byte(v.(string))), nil
+}
+{{- if ne $.ResourceMetadata.Compiler "terraformgoogleconversion-codegen" }}
+func expandSecretManagerSecretVersionPayloadSecretDataWo(v interface{}, d *schema.ResourceData, config *transport_tpg.Config) (interface{}, error) {
+	path := cty.GetAttrPath("secret_data_wo")
+	woVal, _ := d.GetRawConfigAt(path)
+	if !woVal.Type().Equals(cty.String) {
+		return nil, nil
+	}
+	if d.Get("is_secret_data_base64").(bool) {
+		return woVal.AsString(), nil
+	}
+	return base64.StdEncoding.EncodeToString([]byte(woVal.AsString())), nil
```

Comment:
[Some tests still fail ](https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_EPHEMERALWRITEONLYGA__MM_GOOGLE_PACKAGE_SECRETMANAGER/323040)with the panic: value is null

The log  shows that woVal is null when calling the method `github.com/hashicorp/go-cty/cty.Value.AsString`.

Reply:
latest commit should fix it. tested locally and confirmed it. TeamCity build testing it with new commit: https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_GoogleCloud_EPHEMERALWRITEONLYGA__MM_GOOGLE_PACKAGE_SECRETMANAGER/323043